frontend/src/router/index.ts

```diff
@@ -51,48 +62,85 @@ const router = createRouter({
             name: "project",
             component: () => import("../views/ProjectView.vue"),
             props: (route) => ({ projectId: Number(route.params.projectId) }),
+            meta: {
+                middleware: useCanVisit(
+                    useOrCondition(
+                        useIsStudentOfProjectCondition,
+                        useIsInstructorOfProjectCondition
+                    )
+                ),
+            },
         },
         {
             path: "/project/:projectId(\\d+)/submit",
             name: "onSubmit",
             component: () => import("../views/SubmitView.vue"),
             props: (route) => ({ projectId: Number(route.params.projectId) }),
+            meta: {
+                middleware: useCanVisit(useIsInGroupOfProjectCondition),
+            },
         },
         {
             path: "/project/:projectId(\\d+)/groups",
             name: "groups",
             component: () => import("../views/GroupsView.vue"),
             props: (route) => ({ projectId: Number(route.params.projectId) }),
+            meta: {
+                middleware: useCanVisit(
+                    useOrCondition(
+                        useIsStudentOfProjectCondition,
+                        useIsInstructorOfProjectCondition
+                    )
+                ),
+            },
         },
         {
             path: "/groups/:groupId(\\d+)",
             name: "group",
             component: () => import("../views/GroupView.vue"),
             props: (route) => ({ groupId: Number(route.params.groupId) }),
+            meta: {
+                middleware: useCanVisit(
+                    useOrCondition(useIsInGroupCondition, useIsInstructorOfGroupCondition)
+                ),
+            },
         },
         {
             path: "/project/:projectId(\\d+)/submissions",
             name: "projectSubmissions",
             component: () => import("../views/SubmissionsTeacherView.vue"),
             props: (route) => ({ projectId: Number(route.params.projectId) }),
+            meta: {
+                middleware: useCanVisit(useIsInstructorOfProjectCondition),
```

Comment:
als je teacher bent of admin kun je hier niet aan

Reply:
als `is_teacher` van het vak wel vermoed ik? want dat zou me verbazen als dat niet werkt, als `is_teacher` van een ander vak kan je hier inderdaad niet aan, een prof. moet niet aan de indieningen van vakken waar hij geen les geeft kunnen

Reply:
`is_teacher` van een specifiek vak bestaat toch niet?  gwn als user. Bij de backend kan, indien je is_teacher bent, wel aan de submissions, dit was de bedoeling dacht ik. Bijvoorbeeld als je een vak aanmaakt ben je niet automatisch instructor  (kan wel aangepast worden in backend)

Reply:
Mijn verwoording was misschien niet ideaal. Als je `is_teacher` bent, en instructor van vak 1, kan je uiteraard de indieningen van alle projecten hiervan zien. De indieningen van vak 2, waar je geen instructor van bent, kan je niet zien.
Ook als instructor van vak 1, met `is_teacher` op false kan je de indieningen van projecten van dat vak zien.
Ik zal de andere implementatie in de backend over het hoofd gezien hebben.
De implementatie hier vind ik persoonlijk logischer, en ik ben er vrij zeker van dat Ufora en Dodona ook zo zullen werken.

Reply:
ja is idd logischer maar moet zeker nog aagepast worden in backend